add-database

AndreyAA-code opened the pull request.

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed

AndreyAA-code committed:
film

AndreyAA-code committed:
!!! GLOBAL CHANGE
ADDED CODE FROM MY TEST PROJECT
https://github.com/AndreyAA-code/TestFilmorate
!!!!

AndreyAA-code committed:
add h2 database

AndreyAA-code committed:
add h2 database
finished with H2 and Postman tests

AndreyAA-code committed:
add h2 database
finished with H2 and Postman tests

AndreyAA-code committed:
add h2 database
finished with H2 and Postman tests

AndreyAA-code committed:
add h2 database
finished with H2 and Postman tests
Adding tests to project

AndreyAA-code committed:
add h2 database
finished with H2 and Postman tests
Adding tests to project
moved SQL user queries to constants

AndreyAA-code committed:
add h2 database
finished with H2 and Postman tests
Adding tests to project
moved SQL user queries to constants
moved SQL film queries to constants

AndreyAA-code committed:
add h2 database
finished with H2 and Postman tests
Adding tests to project
moved SQL user queries to constants
moved SQL film queries to constants

AndreyAA-code committed:
checked inMemoryBaseTests

AndreyAA-code committed:
checked inMemoryBaseTests
first test passed

AndreyAA-code committed:
checked inMemoryBaseTests
first test passed

AndreyAA-code committed:
checked inMemoryBaseTests
first test passed
все тесты добавлены и проверены

AndreyAA-code committed:
checked inMemoryBaseTests
first test passed
все тесты добавлены и проверены
the end

AndreyAA-code committed:
checked inMemoryBaseTests
first test passed
все тесты добавлены и проверены
the end

AndreyAA-code committed:
checked inMemoryBaseTests
first test passed
все тесты добавлены и проверены
the end

AndreyAA-code committed:
checked inMemoryBaseTests
first test passed
все тесты добавлены и проверены
the end

AndreyAA-code committed:
Merge remote-tracking branch 'origin/main' into add-database

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java
#	src/main/java/ru/yandex/practicum/filmorate/dal/BaseRepository.java
#	src/main/java/ru/yandex/practicum/filmorate/dal/UserRepository.java
#	src/main/java/ru/yandex/practicum/filmorate/dal/mappers/UserRowMapper.java
#	src/main/java/ru/yandex/practicum/filmorate/dto/UpdateUserRequest.java
#	src/main/java/ru/yandex/practicum/filmorate/exceptions/DataAccessException.java
#	src/main/java/ru/yandex/practicum/filmorate/mapper/UserMapper.java
#	src/main/java/ru/yandex/practicum/filmorate/service/UserService.java
#	src/main/resources/application.properties

AndreyAA-code committed:
checked inMemoryBaseTests
first test passed
все тесты добавлены и проверены
the end

AndreyAA-code committed:
checked inMemoryBaseTests
first test passed
все тесты добавлены и проверены
the end

AndreyAA-code committed:
checked inMemoryBaseTests
first test passed
все тесты добавлены и проверены
the end

AndreyAA-code committed:
checked inMemoryBaseTests
first test passed
все тесты добавлены и проверены
the end

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,49 +1,59 @@
 public class FilmController {
 
-    private final FilmService filmService;
+    public final FilmService filmService;
```
Поля следует указывать как private

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/controller/GenreController.java:
```diff
@@ -0,0 +1,29 @@
+@AllArgsConstructor
+@RequestMapping("/genres")
+public class GenreController {
+    public final FilmService filmService;
```
Поля следует указывать как private

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/controller/MpaController.java:
```diff
@@ -0,0 +1,29 @@
+@RequestMapping("/mpa")
+@AllArgsConstructor
+public class MpaController {
+    FilmService filmService;
```
Поля следует указывать как private

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -1,86 +1,66 @@
-    public UserDto getUserById(@PathVariable("userId") long userId) {
-        return userService.getUserById(userId);
-    }
+    public final UserService userService;
```
Поля следует указывать как private

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/mapper/FilmMapper.java:
```diff
@@ -0,0 +1,25 @@
+import ru.yandex.practicum.filmorate.dto.FilmDto;
+import ru.yandex.practicum.filmorate.model.Film;
+
+@Component
```
Component лишний, если метод статический

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/mapper/FilmMapper.java:
```diff
@@ -0,0 +1,25 @@
+
+public class FilmMapper {
+    public static FilmDto mapToFilmDto(Film film) {
+        FilmDto filmDto = new FilmDto();
```
Можно использовать билдер в качестве сеттинга полей

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/mapper/UserMapper.java:
```diff
@@ -1,49 +1,23 @@
 import ru.yandex.practicum.filmorate.dto.UserDto;
 import ru.yandex.practicum.filmorate.model.User;
 
+@Component
```
Component лишний, если метод статический

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -14,16 +15,13 @@ public class Film {
+    private Long duration;
+    private Set<Long> likes = new HashSet<>();
+    private Mpa mpa;
+    private Set<Genre> genres;
```
Тут тоже можно инициализировать коллекцию

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/repository/DbFilmRepository.java:
```diff
@@ -0,0 +1,235 @@
+@Repository
+@AllArgsConstructor
+@Primary
+
```
Лишний отступ между аннотацией и классом

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/repository/DbFilmRepository.java:
```diff
@@ -0,0 +1,235 @@
+    }
+
+    @Override
+    public Collection<Genre> getGenres() {
```
Метод лучше вынести в отдельный класс `GenreRepository`

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/repository/DbFilmRepository.java:
```diff
@@ -0,0 +1,235 @@
+    }
+
+    @Override
+    public Genre getGenresById(Long id) {
```
Метод лучше вынести в отдельный класс `GenreRepository`

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/repository/DbFilmRepository.java:
```diff
@@ -0,0 +1,235 @@
+    }
+
+    @Override
+    public Collection<Mpa> getMpas() {
```
Метод лучше вынести в отдельный класс `MpaRepository`

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/repository/DbFilmRepository.java:
```diff
@@ -0,0 +1,235 @@
+    }
+
+    @Override
+    public Mpa getMpaById(Long id) {
```
Метод лучше вынести в отдельный класс `MpaRepository`

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/repository/DbFilmRepository.java:
```diff
@@ -0,0 +1,235 @@
+        return mpa;
+    }
+
+    public void checkMpaId(Long id) {
```
Методы check лучше указать как `private`, так как они используются только внутри класса  репозитория

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/repository/DbUserRepository.java:
```diff
@@ -0,0 +1,150 @@
+        return commonFriends;
+    }
+
+    public void checkUserId(Long userId) {
```
Аналогично

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/repository/InMemoryFilmRepository.java:
```diff
@@ -0,0 +1,177 @@
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Repository("InMemoryFilmRepository")
```
Имя бина будет автоматически такое `inMemoryFilmRepository`, указывать новое имя не обязательно

AndreyAA-code replied:
выполнено

slavrikk commented on src/main/resources/schema.sql:
```diff
@@ -1,52 +1,45 @@
+    description VARCHAR(200),
+    release_date DATE CHECK (release_date >= '1895-12-28'),
+    duration BIGINT NOT NULL,
+    mpa BIGINT REFERENCES mpa(id)
```
Лучше имя колонки назвать `mpa_id`

AndreyAA-code replied:
выполнено

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed
all reviewer's comments checked

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed
all reviewer's comments checked

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed
all reviewer's comments checked

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed
all reviewer's comments checked

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed
all reviewer's comments checked

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed
all reviewer's comments checked

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed
all reviewer's comments checked

AndreyAA-code committed:
add DB scheme
Завершено ревью пром задания ФЗ 12
добавить БД H2
добавить схему для создания таблиц
добавить схему для создания словарей - рейтинг, жанр, статус дружбы
add set genres to class Film
add jdbc to pom
add UserRepository
add UserService
add UserController
finish UserController and all logic - postman passed
all reviewer's comments checked
